sympy/solvers/solvers.py

```diff
@@ -2653,22 +2654,45 @@ def _tsolve(eq, sym, **flags):
                     return _solve(lhs - rhs, sym, **flags)
             elif sym not in lhs.exp.free_symbols:
                 return _solve(lhs.base - rhs**(1/lhs.exp), sym, **flags)
-            elif not rhs and sym in lhs.exp.free_symbols:
-                # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
-                # the same place
-                sol_base = _solve(lhs.base, sym, **flags)
-                if not sol_base:
-                    return sol_base  # no solutions to remove so return now
-                return list(ordered(set(sol_base) - set(
-                    _solve(lhs.exp, sym, **flags))))
-            elif (rhs is not S.Zero and
-                        lhs.base.is_positive and
-                        lhs.exp.is_real):
-                return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
-            elif lhs.base == 0 and rhs == 1:
-                return _solve(lhs.exp, sym, **flags)
             else:
-                raise NotImplementedError
+                # sym in lhs.exp.free_symbols
+                if not rhs:
+                    # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
+                    # the same place
+                    sol_base = _solve(lhs.base, sym, **flags)
+                    if not sol_base:
+                        return sol_base  # no solutions to remove so return now
+                    return list(ordered(set(sol_base) - set(
+                    _solve(lhs.exp, sym, **flags))))
+                elif lhs.base.is_positive and lhs.exp.is_real:
+                    return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
+                elif rhs.is_Integer:
+                    if  lhs.base == 0:
+                        return _solve(lhs.exp, sym, **flags)
+                    else:
+                        e0=[]
+                        e1=[]
+                        sol=[]
+                        if rhs==1:
+                            e0 = _solve(lhs.exp, sym, **flags)
+                        x=divisors(rhs)
+                        for i in x:
+                            base_sols = solve(lhs.base - i, sym) + solve(lhs.base + i, sym)
+                            for s in set(base_sols):
+                                if eq.subs(sym, s) == 0:
+                                    sol.append(s)
+                        rewrite = lhs.rewrite(exp)
```

Comment:
@oscarbenjamin ,Before rewrite no solution was there. Solution comes in e1.
You can also check by running the following
```
from sympy.solvers.solvers import _tsolve as tsolve
tsolve(3**(2*x + 5) - 4, x)
```